src/java/org/apache/cassandra/service/opstate/KeyspaceTableOpStatePersister.java

```diff
@@ -117,18 +127,39 @@ private Map<KeyspaceTableKey, Instant> readStateFromFile(File file) throws IOExc
         }
     }
 
-    private boolean writeStateToFile(File file, Map<KeyspaceTableKey, Instant> updatedEntries) throws IOException
+    private boolean writeStateToFile(File file, Map<KeyspaceTableKey, Instant> updatedEntries)
     {
         try
         {
-            OBJECT_MAPPER.writeValue(file, convertMapTypeToStringLong(updatedEntries));
+            atomicWritetoFile(file, OBJECT_MAPPER.writeValueAsString(convertMapTypeToStringLong(updatedEntries)));
+            return true;
+        }
+        catch (IOException e)
+        {
+            return false;
+        }
+    }
+
+    private void atomicWritetoFile(File file, String content) throws IOException
```

Comment:
you probably want to synchronize this too right? otherwise two writes could clobber each other

Reply:
Yep, totally right, especially since we now sometimes write at read time as well